src/main/java/com/faforever/server/ladder/DivisionService.java

```diff
@@ -0,0 +1,144 @@
+package com.faforever.server.ladder;
+
+import com.faforever.server.config.ServerProperties;
+import com.faforever.server.entity.Division;
+import com.faforever.server.entity.Player;
+import com.faforever.server.entity.PlayerDivisionInfo;
+import com.google.common.collect.ImmutableList;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.springframework.stereotype.Service;
+
+import javax.annotation.PostConstruct;
+import javax.inject.Inject;
+import javax.transaction.Transactional;
+import java.text.MessageFormat;
+import java.util.Objects;
+
+@Service
+@Slf4j
+public class DivisionService {
+  private final ServerProperties properties;
+  private final DivisionRepository divisionRepository;
+  private final PlayerDivisionInfoRepository playerDivisionInfoRepository;
+
+  @Getter
+  private ImmutableList<Division> divisions;
+
+  @Inject
+  public DivisionService(ServerProperties properties, DivisionRepository divisionRepository, PlayerDivisionInfoRepository playerDivisionInfoRepository) {
+    this.properties = properties;
+    this.divisionRepository = divisionRepository;
+    this.playerDivisionInfoRepository = playerDivisionInfoRepository;
+  }
+
+  @PostConstruct
+  public void init() {
+    divisions = ImmutableList.copyOf(divisionRepository.findAllByOrderByLeagueAscThresholdAsc());
+  }
+
+  public Division getCurrentPlayerDivision(Player player) {
+    int season = properties.getLadder().getSeason();
+
+    PlayerDivisionInfo info = playerDivisionInfoRepository.findByPlayerAndSeason(player, season);
+
+    if(info == null) {
+      return null;
+    }
+
+    return divisions.stream()
+      .filter(division -> Objects.equals(division.getLeague(), info.getLeague()))
+      .filter(division -> info.getScore() <= division.getThreshold())
+      .findFirst()
+      .orElseThrow(() -> new IllegalStateException(MessageFormat.format("Could not determine division for PlayerDivisionInfo {0}", info)));
+  }
+
+  @Transactional
+  public void postResult(@NotNull Player playerOne, @NotNull Player playerTwo, @Nullable Player winner) {
+    int season = properties.getLadder().getSeason();
+
+    PlayerDivisionInfo playerOneInfo = getPlayerDivisionInfo(playerOne, season);
+    PlayerDivisionInfo playerTwoInfo = getPlayerDivisionInfo(playerTwo, season);
+
+    playerOneInfo.setGames(playerOneInfo.getGames() + 1);
+    playerTwoInfo.setGames(playerTwoInfo.getGames() + 1);
+
+    if (winner == null) {
+      log.info("Game ended in a draw - no changes in score");
```

Comment:
Reduce log level to `debug`. See https://github.com/FAForever/java-guidelines/wiki/Contribution-Guidelines#logging